hasher-matcher-actioner/hmalib/common/actioner_models.py

```diff
@@ -158,118 +158,69 @@ def from_match_message_and_label(
 
 
 @dataclass
-class ActionPerformer:
+class ActionPerformer(config.HMAConfigWithSubtypes):
     """
-    A configuration of what action should be performed when a specific label
-    is added to a match. Each label can lead to exactly one action being taken.
+    An ActionPerfomer is the configuration + the code to perform an action.
 
-    To create a new type of ActionPerformer simply extend this class
-    and add the necessary fields as attributes. All these attributes will be stored in AWS
-    after update_aws_config is called. To retrieve from AWS the correct
-    action performer for a label use the function ActionPerformerConfig.get_performer
+    All actions share the same namespace (so that a post action and a
+    "send to review" action can't both be called "IActionReview")
 
-    NOTE all attributes must be aws serializable. see config._py_to_aws_field for serialization logic
+    ActionPerformer.get("action_name").perform_action(match_message)
     """
 
-    action_label: ActionLabel
+    @staticmethod
+    def get_subtype_classes():
+        return [
+            WebhookPostActionPerformer,
+            WebhookGetActionPerformer,
+            WebhookPutActionPerformer,
+            WebhookDeleteActionPerformer,
```

Comment:
Decided to keep these separate?

Reply:
Yea, asked you about it in chat, and wanted to avoid messing up any UI work you had in flight. Should be easy to compress later if you decide to.